src/core/iomgr/fd_posix.h

```diff
@@ -111,7 +112,7 @@ grpc_fd *grpc_fd_create(int fd, const char *name);
    notify_on_write.
    MUST NOT be called with a pollset lock taken */
 void grpc_fd_orphan(grpc_exec_ctx *exec_ctx, grpc_fd *fd, grpc_closure *on_done,
```

Comment:
How about:

```c
void grpc_fd_orphan(..., int *release_fd, ...);
```

With the semantics:
if release_fd == NULL, closes the underlying file descriptor
if release_fd != NULL, does not close the file descriptor, but set *release_fd = underlying_fd before calling on_done